tests/output.rs

```diff
@@ -1,228 +1,118 @@
-) {
-    Command::cargo_bin("tree-sitter-grep")
+fn assert_sorted_output(fixture_dir_name: &str, command_and_output: &str) {
+    let CommandAndOutput {
+        mut command_line_args,
+        output,
+    } = parse_command_and_output(command_and_output);
+    let command_name = command_line_args.remove(0);
+    Command::cargo_bin(&command_name)
         .unwrap()
-        .args(args)
-        .current_dir(current_dir)
+        .args(command_line_args)
+        .current_dir(get_fixture_dir_path_from_name(fixture_dir_name))
         .assert()
         .success()
-        .stdout(predicate_from_expected_matches(
-            expected_matches,
-            output_mode,
-        ));
-}
-
-#[derive(Copy, Clone, Debug, Eq, PartialEq)]
-enum OutputMode {
-    Normal,
-    Vimgrep,
-}
-
-impl OutputMode {
-    pub fn get_expected_total_number_of_match_lines(
-        &self,
-        expected_matches: &[ExpectedMatch],
-    ) -> usize {
-        match self {
-            Self::Normal => expected_matches
-                .into_iter()
-                .map(|expected_match| expected_match.lines.len())
-                .sum(),
-            Self::Vimgrep => expected_matches.len(),
-        }
-    }
-
-    pub fn expected_output_text(&self, expected_match: &ExpectedMatch) -> String {
-        match self {
-            Self::Normal => expected_match
-                .lines
-                .iter()
-                .enumerate()
-                .map(|(line_index, line)| {
-                    format!(
-                        "{}:{}:{}",
-                        with_leading_dot_slash(&expected_match.relative_file_path),
-                        expected_match.range.start.line + line_index + 1,
-                        line
-                    )
-                })
-                .collect::<Vec<_>>()
-                .join("\n"),
-            Self::Vimgrep => format!(
-                "{}:{}:{}:{}",
-                expected_match.relative_file_path,
-                expected_match.range.start.line + 1,
-                expected_match.range.start.column + 1,
-                &expected_match.lines[0]
-            ),
-        }
-    }
-}
-
-fn with_leading_dot_slash(relative_path: &str) -> String {
-    if relative_path.starts_with(".") {
-        relative_path.to_owned()
-    } else {
-        format!("./{relative_path}")
-    }
+        .stdout(predicate::function(|actual_output| {
+            do_sorted_lines_match(actual_output, &output)
+        }));
 }
 
-struct FixtureQueryExpectedMatches {
-    pub fixture_dir_name: String,
-    pub language: String,
-    pub query_source: String,
-    pub expected_matches: Vec<ExpectedMatch>,
+fn do_sorted_lines_match(actual_output: &str, expected_output: &str) -> bool {
+    let mut actual_lines = actual_output.split("\n").collect::<Vec<_>>();
+    actual_lines.sort();
+    let mut expected_lines = expected_output.split("\n").collect::<Vec<_>>();
+    expected_lines.sort();
+    actual_lines == expected_lines
```

Comment:
This isn't comparing the lines in some "canonical sensible" order, by doing lexicographic sorting here you'd get things like a match on line 12 coming before a match on line 3 etc, but if all of the individual sorted lines match we're probably in good shape unless we're worried about eg multi-line matches not being output in the correct order wrt each other?

Reply:
> unless we're worried about eg multi-line matches not being output in the correct order wrt each other

Like if there were overlapping matches, you might get 2 different matches from the same line in the same file? (Which would be weird to sort if they're multi-line matches?)

This seems good enough ™️ to me for now.

Reply:
Ya technically my understanding would be "the only way this would give a false positive (where "positive" == "passed") is if what was actually output was the same exact set of lines just in a different order (but where that different order wasn't acceptable ie wasn't just a different output order of the matches)", which seems like the only surface area there is "some multi-line match wasn't printed out contiguously"

I agree it seems like significantly more work to try and test that and I'd guess we either have interleaving problems or we don't